src/ints/mouse.cpp

```diff
@@ -58,118 +97,150 @@ struct button_event {
+                                              0x003F,
+                                              0x001F,
+                                              0x01FF,
+                                              0x00FF,
+                                              0x30FF,
+                                              0xF87F,
+                                              0xF87F,
+                                              0xFCFF};
+
+static uint16_t defaultCursorMask[CURSORY] = {0x0000,
+                                              0x4000,
+                                              0x6000,
+                                              0x7000,
+                                              0x7800,
+                                              0x7C00,
+                                              0x7E00,
+                                              0x7F00,
+                                              0x7F80,
+                                              0x7C00,
+                                              0x6C00,
+                                              0x4600,
+                                              0x0600,
+                                              0x0300,
+                                              0x0300,
+                                              0x0000};
 
 static uint16_t userdefScreenMask[CURSORY];
 static uint16_t userdefCursorMask[CURSORY];
 
 static struct {
-	uint8_t buttons;
-	uint16_t times_pressed[MOUSE_BUTTONS];
-	uint16_t times_released[MOUSE_BUTTONS];
-	uint16_t last_released_x[MOUSE_BUTTONS];
-	uint16_t last_released_y[MOUSE_BUTTONS];
-	uint16_t last_pressed_x[MOUSE_BUTTONS];
-	uint16_t last_pressed_y[MOUSE_BUTTONS];
-	uint16_t hidden;
-	float add_x,add_y;
-	int16_t min_x,max_x,min_y,max_y;
-	float mickey_x,mickey_y;
-	float x,y;
-	button_event event_queue[QUEUE_SIZE];
-	uint8_t events;//Increase if QUEUE_SIZE >255 (currently 32)
-	uint16_t sub_seg,sub_ofs;
-	uint16_t sub_mask;
-
-	bool	background;
-	int16_t	backposx, backposy;
-	uint8_t	backData[CURSORX*CURSORY];
-	uint16_t*	screenMask;
-	uint16_t* cursorMask;
-	int16_t	clipx,clipy;
-	int16_t  hotx,hoty;
-	uint16_t  textAndMask, textXorMask;
-
-	float	mickeysPerPixel_x;
-	float	mickeysPerPixel_y;
-	float	pixelPerMickey_x;
-	float	pixelPerMickey_y;
-	uint16_t	senv_x_val;
-	uint16_t	senv_y_val;
-	uint16_t	dspeed_val;
-	float	senv_x;
-	float	senv_y;
-	int16_t  updateRegion_x[2];
-	int16_t  updateRegion_y[2];
-	uint16_t  doubleSpeedThreshold;
-	uint16_t  language;
-	uint16_t  cursorType;
-	uint16_t	oldhidden;
-	uint8_t  page;
-	bool enabled;
-	bool inhibit_draw;
-	bool timer_in_progress;
-	bool in_UIR;
-	uint8_t mode;
-	int16_t gran_x,gran_y;
+
+    // TODO - DANGER, WILL ROBINSON!
+    // This whole structure can be read or written from the guest side via virtual DOS driver,
+    // functions 0x15 / 0x16 / 0x17; we need to make sure nothing can be broken by malicious code!
+
+    uint16_t times_pressed[MOUSE_BUTTONS];
```

Comment:
Assign default values to all members and put members on their own lines.

Can use empty lines to "block-format" each group of related variables together, chunk by chunk :+1:

Reply:
Done.